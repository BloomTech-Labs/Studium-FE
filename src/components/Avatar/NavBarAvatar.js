import React, {useState} from 'react';
import {Avatar} from 'antd';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import {Popover} from 'antd';
import {signOut} from '../../actions';
import {APP_PATHS} from '../../utilities/constants.js';

/**
 * Nav Bar Avatar
 * @component
 * @example
 * return (
 *  <NavBarAvatar />
 *  )
 */
export const NavBarAvatar = ({getHooks, avatarUrl, ...props}) => {
  
  const [open, setOpen] = useState(false);
  const {dispatch, usersState, changePath} = getHooks();
  
  const handleClick = (path) => {
    setOpen(false);
    if(path === 'logout'){
      dispatch(signOut());
    }else if(path === 'signin'){
      changePath(APP_PATHS.SIGN_IN_PATH);
    }
  };
  
  const getContent = () => {
    return (
      <AvatarMenu>
        {/*<p onClick={() => handleClick("logout")}>Edit Profile</p>*/}
        <p style={{background: '#D7EDE8'}}
           onClick={() => handleClick('logout')}>Logout</p>
      </AvatarMenu>
    );
  };
  
  return (
    <Popover placement="bottomRight"
             title={'Hi ' + usersState.user.displayName}
             content={getContent()}
             visible={open}
             trigger="click">
      {avatarUrl ? <StyledAntAvatar src={avatarUrl} {...props} size={40}
                                    onClick={() => setOpen(!open)}
        /> :
        <StyledAntAvatar {...props} size={40} onClick={() => setOpen(!open)}
        />}
    </Popover>
  );
};

const StyledAntAvatar = styled(Avatar)`
  && {
    position: absolute;
    background-color: #585858;
    filter: contrast(0.5);
    top: 35px;
    right: 8%;
    transform: translate(0, -50%);
  }
`;

const AvatarMenu = styled.div`
background: #A8D6CC;
width: 100%;
p {
  padding: 10px;
  font-weight: bold;
  color: #00352A;
}
`;

NavBarAvatar.propTypes = {
  getHooks: PropTypes.func,
  avatarUrl: PropTypes.string,
};
