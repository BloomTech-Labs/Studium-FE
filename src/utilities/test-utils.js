import React from 'react';
import {render} from '@testing-library/react';
import AppProvider from './AppProviders.js';
import {getStore} from './getStore.js';

const store = getStore();

/**
 * Custom Render
 * @category Utilities
 *
 * @description Custom render function to wrap our components with provides
 * they need for testing purposes.
 *
 * @param ui
 * @param options
 * @return {CustomRenderResults}
 *
 */
export const customRender = (ui, options) => render(ui, {
  wrapper: AppProvider, ...options,
});

/**
 * Get Child Nodes
 * @category Utilities
 *
 * @description Gets all the child nodes from a HtmlElment for react testing
 * library.
 *
 * @param {HTMLElement} c
 * @return {HTMLElement[]}
 */
export const getChildNodes = c => {
  console.log('inside of get next node.');
  const childNodes = [];
  if(c.hasChildNodes()){
    c.childNodes.forEach(child => {
      childNodes.push(child);
    });
  }
  return childNodes;
};

/**
 * Get Nodes By Node Type
 * @category Utilities
 *
 * @description Returns all the child HtmlElements that are of the requested
 * type.
 *
 * @param {HTMLElement} c
 * @param {string} type
 * @return {HTMLElement[]}
 */
export const getNodesByType = (c, type) => {
  const stack = [];
  const toReturn = [];
  stack.push(c);
  while(stack.length > 0){
    let node = stack.pop();
    if(node.hasChildNodes()){
      node.childNodes.forEach(child => {
        stack.push(child);
      });
    }
    if(node.nodeName === type.toUpperCase()){
      toReturn.push(node);
    }
  }
  
  return toReturn;
};

export * from '@testing-library/react';

export {store};

/**
 * @typedef {object} CustomRenderResults
 * @property {HTMLDivElement} container
 * @property {HTMLBodyElement} baseElement
 * @property {function} debug
 * @property {function} unmount
 * @property {function} rerender
 * @property {function} asFragment
 * @property {function} queryAllByLabelText
 * @property {function} getAllByLabelText
 * @property {function} queryByLabelText
 * @property {function} getAllByLabelText
 * @property {function} queryByLabelText
 * @property {function} getByLabelText
 * @property {function} findAllByLabelText
 * @property {function} findByLabelText
 * @property {function} queryByPlaceholderText
 * @property {function} queryAllByPlaceholderText
 * @property {function} getByPlaceholderText
 * @property {function} getAllByPlaceholderText
 * @property {function} findAllByPlaceholderText
 * @property {function} findByPlaceholderText
 * @property {function} queryAllByText
 * @property {function} queryByText
 * @property {function} getByText
 * @property {function} getAllByText
 * @property {function} findAllByText
 * @property {function} findByText
 * @property {function} queryAllByDisplayValue
 * @property {function} queryByDisplayValue
 * @property {function} getByDisplayValue
 * @property {function} getAllByDisplayValue
 * @property {function} findAllByDisplayValue
 * @property {function} findByDisplayValue
 * @property {function} queryAllByAltText
 * @property {function} queryByAltText
 * @property {function} getByAltText
 * @property {function} getAllByAltText
 * @property {function} findAllByAltText
 * @property {function} findByAltText
 * @property {function} queryAllByTitle
 * @property {function} queryByTitle
 * @property {function} getByTitle
 * @property {function} getAllByTitle
 * @property {function} findAllByTitle
 * @property {function} findByTitle
 * @property {function} queryAllByRole
 * @property {function} queryByRole
 * @property {function} getAllByRole
 * @property {function} getByRole
 * @property {function} findAllByRole
 * @property {function} findByRole
 * @property {function} queryByTestId
 * @property {function} queryAllByTestId
 * @property {function} getByTestId
 * @property {function} getAllByTestId
 * @property {function} findAllByTestId
 * @property {function} findByTestId
 */