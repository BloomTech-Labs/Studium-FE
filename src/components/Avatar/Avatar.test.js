import React from 'react';
import renderer from 'react-test-renderer';
import {customRender} from '../../utilities/test-utils.js';

import {NavBarAvatar} from './NavBarAvatar.js';

test('StyledAvatar render correctly', () => {
  const {container} = customRender(<NavBarAvatar/>);
  expect(container).toMatchSnapshot();
});