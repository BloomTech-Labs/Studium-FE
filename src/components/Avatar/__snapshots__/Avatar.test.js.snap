// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledAvatar render correctly 1`] = `
.c0.c0 {
  position: absolute;
  background-color: #585858;
  -webkit-filter: contrast(0.5);
  filter: contrast(0.5);
  top: 35px;
  right: 8%;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
}

<div>
   
  <span
    class="ant-avatar c0 ant-avatar-circle"
    style="width: 40px; height: 40px; line-height: 40px; font-size: 18px;"
  >
    <span
      class="ant-avatar-string"
      style="line-height: 40px; transform: scale(1) translateX(-50%);"
    />
  </span>
   
</div>
`;
